x-pack/plugins/snapshot_restore/server/types.ts

```diff
@@ -31,4 +31,20 @@ export interface RouteDependencies {
   };
 }
 
+/**
+ * An object representing a resolved index, data stream or alias
+ */
+interface DataObjectFromES {
```

Comment:
The ES engineers have called these "Index abstractions". We could reuse that term here for consistency if you want.